### ISSUE TITLE
Per page limit is not working with Elementor Loop and current query

### ISSUE THREAD

**Issue by jagdish1o1** [elementor/elementor#21798]
### Prerequisites

- [X] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [X] The issue still exists against the latest stable version of Elementor.

### Description

Hello team, 

I think it's a bug, when I'm creating a loop with current query for category pages and limiting it to only one post per page, it's totally ignoring it and rendering general setting from wp. 

I even tried using the custom query filter and try that way but no luck. 

`
    function register() {
        add_action( 'elementor/query/chs_feature_1', array($this, 'chs_feature_1_callback') );

    }

    function chs_feature_1_callback($query) {

        $query->set( 'posts_per_page', 1 );

    }
`

### Steps to reproduce

Simply create the loop for category archive and use current query and set the limit to 1. 

### Isolating the problem

- [X] This bug happens with only Elementor plugin active (and Elementor Pro).
- [X] This bug happens with a Blank WordPress theme active ([Hello theme](https://wordpress.org/themes/hello-elementor/)).
- [X] I can reproduce this bug consistently following the steps above.

### System Info

					
== Server Environment ==
	Operating System: Linux
	Software: Apache/2.4
	MySQL version: MariaDB Server (GPL), Release 10.5.17.1.bolt v10.5.17
	PHP Version: 8.1.14
	PHP Memory Limit: 64M
	PHP Max Input Vars: 1000
	PHP Max Post Size: 50M
	GD Installed: Yes
	ZIP Installed: Yes
	Write Permissions: All right
	Elementor Library: Connected

== WordPress Environment ==
	Version: 6.1.1
	Site URL: https://weltme-19342.siv-ams.servebolt.cloud
	Home URL: https://weltme-19342.siv-ams.servebolt.cloud
	WP Multisite: No
	Max Upload Size: 50 MB
	Memory limit: 128M
	Max Memory limit: 128M
	Permalink Structure: /%year%/%monthnum%/%day%/%postname%/
	Language: en-US
	Timezone: 0
	Admin Email: jagdish@riseable.in
	Debug Mode: Inactive

== Theme ==
	Name: Hello Elementor
	Version: 2.6.1
	Author: Elementor Team
	Child Theme: No

== User ==
	Role: administrator
	WP Profile lang: en_US
	User Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/111.0.0.0 Safari/537.36

== Active Plugins ==
	Advanced Ads
		Version: 1.41.0
		Author: Thomas Maier, Advanced Ads GmbH

	Advanced Ads Pro
		Version: 2.21.2
		Author: Advanced Ads GmbH

	Advanced Ads – PopUp and Layer Ads
		Version: 1.7.7
		Author: Advanced Ads GmbH

	Advanced Custom Fields PRO
		Version: 6.0.7
		Author: WP Engine

	Better Elementor Excerpt
		Version: 1.0.0
		Author: Patryk Kożuch

	Converter for Media
		Version: 5.8.4
		Author: matt plugins

	Duplicate Page
		Version: 4.5.1
		Author: mndpsingh287

	Easy Table of Contents
		Version: 2.0.46
		Author: Magazine3

	Elementor
		Version: 3.11.5
		Author: Elementor.com

	Elementor Pro
		Version: 3.11.7
		Author: Elementor.com

	Forestlake Customization
		Version: 1.0
		Author: Jagdish

	Smart Slider 3
		Version: 3.5.1.14
		Author: Nextend

	Smush
		Version: 3.12.6
		Author: WPMU DEV

== Elements Usage ==
	
	archive : 1
		container : 5
		loop-carousel : 1
		loop-grid : 2
	footer : 1
		form : 1
		heading : 3
		icon-list : 1
		social-icons : 1
		spacer : 1
		text-editor : 1
	header : 2
		heading : 1
		nav-menu : 2
	section : 1
		heading : 1
		shortcode : 1
	widget : 0
		image-box : 1
	wp-page : 1
		heading : 5
		html : 1
		loop-carousel : 1
		loop-grid : 4
		shortcode : 2
		smartslider : 1
	single-post : 3
		container : 14
		heading : 4
		loop-grid : 1
		shortcode : 5
		theme-post-content : 1
		theme-post-featured-image : 1
		theme-post-title : 1
	loop-item : 10
		better-post-excerpt : 3
		container : 9
		image : 1
		post-info : 4
		theme-post-excerpt : 1
		theme-post-featured-image : 7
		theme-post-title : 10

== Settings ==
	
	cpt_support: post, page
	allow_tracking: yes
	css_print_method: internal
	font_display: swap

== Features ==
	Custom Fonts: 1
	Custom Icons: 0

== Integrations ==
	

== Elementor Experiments ==
	Optimized DOM Output: Active by default
	Improved Asset Loading: Active by default
	Improved CSS Loading: Active by default
	Inline Font Icons: Active
	Accessibility Improvements: Active by default
	Additional Custom Breakpoints: Active by default
	admin_menu_rearrangement: Inactive by default
	Flexbox Container: Active
	Upgrade Swiper Library: Active by default
	Default to New Theme Builder: Active by default
	Hello Theme Header & Footer: Active by default
	Editor V2: Inactive by default
	Landing Pages: Active by default
	Nested Elements: Inactive by default
	Lazy Load Background Images: Inactive by default
	Save as Default: Active by default
	Page Transitions: Active by default
	Notes: Active by default
	Loop: Active by default
	Form Submissions: Active by default
	Scroll Snap: Active by default
	Mega Menu: Inactive by default

== Log ==
	
PHP: showing 9 of 9PHP: 2023-03-10 09:29:50 [notice X 13722][/kunder/gogett_12852/weltme_19342/public/wp-content/plugins/elementor-pro/modules/loop-builder/module.php::200] strpos(): Passing null to parameter #1 ($haystack) of type string is deprecated [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2023-03-10 09:30:12 [warning X 28][/kunder/gogett_12852/weltme_19342/public/wp-content/plugins/elementor/core/common/modules/ajax/module.php::175] Undefined array key "data" [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2023-03-10 10:58:03 [error X 8][/kunder/gogett_12852/weltme_19342/public/wp-content/plugins/elementor-pro/modules/theme-builder/classes/conditions-manager.php::295] Uncaught Error: Call to a member function update_meta() on bool in /kunder/gogett_12852/weltme_19342/public/wp-content/plugins/elementor-pro/modules/theme-builder/classes/conditions-manager.php:295
Stack trace:
#0 /kunder/gogett_12852/weltme_19342/public/wp-content/plugins/elementor-pro/modules/theme-builder/classes/conditions-manager.php(168): ElementorPro\Modules\ThemeBuilder\Classes\Conditions_Manager->save_conditions()
#1 [internal function]: ElementorPro\Modules\ThemeBuilder\Classes\Conditions_Manager->ajax_save_theme_template_conditions()
#2 /kunder/gogett_12852/weltme_19342/public/wp-content/plugins/elementor/core/common/modules/ajax/module.php(175): call_user_func()
#3 /kunder/gogett_12852/weltme_19342/public/wp-includes/class-wp-hook.php(308): Elementor\Core\Common\Modules\Ajax\Module->handle_ajax_request()
#4 /kunder/gogett_12852/weltme_19342/public/wp-includes/class-wp-hook.php(332): WP_Hook->apply_filters()
#5 /kunder/gogett_12852/weltme_19342/public/wp-includes/plugin.php(517): WP_Hook->do_action()
#6 /kunder/gogett_12852/weltme_19342/public/wp-admin/admin-ajax.php(188): do_action()
#7 {main}
  thrown [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2023-03-11 05:02:33 [notice X 41][/kunder/gogett_12852/weltme_19342/public/wp-content/plugins/elementor-pro/modules/custom-css/module.php::76] trim(): Passing null to parameter #1 ($string) of type string is deprecated [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2023-03-11 05:14:22 [notice X 79][/kunder/gogett_12852/weltme_19342/public/wp-content/plugins/elementor-pro/modules/loop-builder/module.php::200] strpos(): Passing null to parameter #1 ($haystack) of type string is deprecated [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> rest_error_handler()
#1: /kunder/gogett_12852/weltme_19342/public/wp-content/plugins/elementor-pro/modules/loop-builder/module.php(200): class type strpos()
#2: /kunder/gogett_12852/weltme_19342/public/wp-content/plugins/elementor-pro/modules/loop-builder/module.php(212): ElementorPro\Modules\LoopBuilder\Module -> is_editing_existing_loop_item()
#3: /kunder/gogett_12852/weltme_19342/public/wp-content/plugins/elementor-pro/modules/loop-builder/module.php(60): ElementorPro\Modules\LoopBuilder\Module -> is_loop_theme_builder()
#4: /kunder/gogett_12852/weltme_19342/public/wp-content/plugins/elementor-pro/modules/loop-builder/documents/loop.php(386): ElementorPro\Modules\LoopBuilder\Module -> __construct()
',
)]
PHP: 2023-03-11 05:14:22 [notice X 79][/kunder/gogett_12852/weltme_19342/public/wp-content/plugins/elementor-pro/modules/loop-builder/module.php::200] strpos(): Passing null to parameter #1 ($haystack) of type string is deprecated [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> rest_error_handler()
#1: /kunder/gogett_12852/weltme_19342/public/wp-content/plugins/elementor-pro/modules/loop-builder/module.php(200): class type strpos()
#2: /kunder/gogett_12852/weltme_19342/public/wp-content/plugins/elementor-pro/modules/loop-builder/module.php(212): ElementorPro\Modules\LoopBuilder\Module -> is_editing_existing_loop_item()
#3: /kunder/gogett_12852/weltme_19342/public/wp-content/plugins/elementor-pro/modules/loop-builder/module.php(60): ElementorPro\Modules\LoopBuilder\Module -> is_loop_theme_builder()
#4: /kunder/gogett_12852/weltme_19342/public/wp-content/plugins/elementor-pro/modules/loop-builder/documents/loop.php(308): ElementorPro\Modules\LoopBuilder\Module -> __construct()
',
)]
PHP: 2023-03-11 05:38:54 [warning X 69][/kunder/gogett_12852/weltme_19342/public/wp-content/plugins/elementor-pro/core/app/modules/site-editor/data/endpoints/templates.php::150] Undefined array key "condition_type" [array (
  'trace' => '
#0: /kunder/gogett_12852/weltme_19342/public/wp-content/plugins/elementor-pro/core/app/modules/site-editor/data/endpoints/templates.php(150): Elementor\Core\Logger\Manager -> rest_error_handler()
#1: ElementorPro\Core\App\Modules\SiteEditor\Data\Endpoints\Templates -> normalize_template_json_item()
#2: /kunder/gogett_12852/weltme_19342/public/wp-content/plugins/elementor-pro/core/app/modules/site-editor/data/endpoints/templates.php(120): class type array_map()
#3: /kunder/gogett_12852/weltme_19342/public/wp-content/plugins/elementor-pro/core/app/modules/site-editor/data/endpoints/templates.php(59): ElementorPro\Core\App\Modules\SiteEditor\Data\Endpoints\Templates -> normalize_templates_json()
#4: /kunder/gogett_12852/weltme_19342/public/wp-content/plugins/elementor/data/base/endpoint.php(158): ElementorPro\Core\App\Modules\SiteEditor\Data\Endpoints\Templates -> get_items()
',
)]
PHP: 2023-03-11 05:50:50 [warning X 9][/kunder/gogett_12852/weltme_19342/public/wp-content/plugins/elementor-pro/modules/theme-builder/classes/conditions-manager.php::168] Undefined array key "editor_post_id" [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2023-03-14 07:40:49 [warning X 1][/kunder/gogett_12852/weltme_19342/public/wp-content/plugins/elementor-pro/modules/dynamic-tags/acf/tags/acf-image.php::40] Undefined array key 1 [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]

JS: showing 8 of 8JS: 2023-03-10 10:31:45 [error X 59][https://weltme-19342.siv-ams.servebolt.cloud/wp-admin/load-scripts.php?c=0&loadchunk_0=underscore,jquery-core,jquery-migrate,backbone,jquery-ui-core,jquery-ui-mouse,wp-api-request,regenerator-runtime,wp-polyfill,wp-&loadchunk_1=hooks&ver=6.1.1:4:51824] Cannot read properties of undefined (reading 'replace') 
JS: 2023-03-10 10:53:59 [error X 1][https://weltme-19342.siv-ams.servebolt.cloud/wp-content/plugins/elementor/assets/js/editor.min.js?ver=3.11.3:3:899294] elementorFrontend is not defined 
JS: 2023-03-10 11:25:19 [error X 12][https://weltme-19342.siv-ams.servebolt.cloud/wp-content/plugins/elementor/assets/js/editor.min.js?ver=3.11.3:3:1087291] Cannot read properties of null (reading 'getBoundingClientRect') 
JS: 2023-03-11 06:59:44 [error X 1][https://weltme-19342.siv-ams.servebolt.cloud/wp-content/plugins/elementor/assets/js/editor.min.js?ver=3.11.3:3:670939] this.model.isValidChild is not a function 
JS: 2023-03-13 07:52:05 [error X 3][https://weltme-19342.siv-ams.servebolt.cloud/wp-content/plugins/elementor/assets/js/editor.min.js?ver=3.11.3:3:791668] Cannot read properties of undefined (reading 'localized') 
JS: 2023-03-13 14:23:26 [error X 1][https://weltme-19342.siv-ams.servebolt.cloud/wp-content/plugins/elementor-pro/assets/js/editor.min.js?ver=3.11.5:2:61404] Cannot convert undefined or null to object 
JS: 2023-03-14 07:14:11 [error X 6][https://weltme-19342.siv-ams.servebolt.cloud/wp-content/plugins/elementor-pro/assets/lib/sticky/jquery.sticky.min.js?ver=3.11.5:1:1162] Failed to execute 'getComputedStyle' on 'Window': parameter 1 is not of type 'Element'. 
JS: 2023-03-15 10:08:26 [error X 5][https://weltme-19342.siv-ams.servebolt.cloud/wp-content/plugins/elementor/assets/js/common-modules.min.js?ver=3.11.5:2:53736] Model.isValidChild() should be implemented, please provide 'isValidChild' functionality. 

Log: showing 20 of 202023-03-13 13:38:01 [info] elementor::elementor_updater Started 
2023-03-13 13:38:01 [info] Elementor/Upgrades - _on_each_version Start  
2023-03-13 13:38:01 [info] Elementor/Upgrades - _on_each_version Finished 
2023-03-13 13:38:01 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.11.3',
  'to' => '3.11.4',
)]
2023-03-13 13:38:01 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '3.11.3',
  'to' => '3.11.4',
)]
2023-03-13 13:38:06 [info] elementor-pro::elementor_pro_updater Started 
2023-03-13 13:38:06 [info] Elementor Pro/Upgrades - _on_each_version Start  
2023-03-13 13:38:06 [info] Elementor Pro/Upgrades - _on_each_version Finished 
2023-03-13 13:38:06 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.11.4',
  'to' => '3.11.5',
)]
2023-03-13 13:38:06 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.11.4',
  'to' => '3.11.5',
)]
2023-03-15 06:58:13 [info] elementor-pro::elementor_pro_updater Started 
2023-03-15 06:58:13 [info] Elementor Pro/Upgrades - _on_each_version Start  
2023-03-15 06:58:13 [info] Elementor Pro/Upgrades - _on_each_version Finished 
2023-03-15 06:58:13 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.11.5',
  'to' => '3.11.6',
)]
2023-03-15 06:58:13 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.11.5',
  'to' => '3.11.6',
)]
2023-03-15 06:58:13 [info] elementor::elementor_updater Started 
2023-03-15 06:58:13 [info] Elementor/Upgrades - _on_each_version Start  
2023-03-15 06:58:13 [info] Elementor/Upgrades - _on_each_version Finished 
2023-03-15 06:58:13 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.11.4',
  'to' => '3.11.5',
)]
2023-03-15 06:58:13 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '3.11.4',
  'to' => '3.11.5',
)]

== Elementor - Compatibility Tag ==
	
	Better Elementor Excerpt: Compatibility not specified
	Elementor Pro: Compatible

== Elementor Pro - Compatibility Tag ==

**Comment by vladninja** [comment]
@nicholaszein It looks like a simple fix, but there has been no progress from March 2023
